File: abstractions/java/lib/src/main/java/com/microsoft/kiota/serialization/Parsable.java
package com.microsoft.kiota.serialization;

import java.util.Map;
import java.util.function.Consumer;

import javax.annotation.Nonnull;
/**
 * Defines a serializable model object.
 */
public interface Parsable {
    /**
     * Gets the deserialization information for this object.
     * @return The deserialization information for this object where each entry is a property key with its deserialization callback.
     */
    @Nonnull
    Map<String, Consumer<ParseNode>> getFieldDeserializers();
    /**
     * Writes the objects properties to the current writer.
     * @param writer The writer to write to.
     */
    void serialize(@Nonnull final SerializationWriter writer);
}

File: abstractions/php/src/Serialization/Parsable.php
<?php
namespace Microsoft\Kiota\Abstractions\Serialization;

/**
 * Defines the serializable model object.
 */
interface Parsable {

    /**
     * Gets the deserialization information for this object.
     * @return array<string, callable> The deserialization information for this object where each entry is a property key with its deserialization callback.
     */
    public function getFieldDeserializers(): array;

    /**
     * Writes the objects properties to the current writer.
     * @param SerializationWriter $writer The writer to write to.
     */
    public function serialize(SerializationWriter $writer): void;
}
